.github/workflows/ci.yml

```diff
@@ -36,9 +36,10 @@ jobs:
         path: |
           {{ matrix.path }}
           .tox
-        key: ${{ runner.os }}-venvs-${{ hashFiles('**/poetry.lock') }}
+        key: ${{ matrix.os }}-venvs-${{ matrix.python-version }}-${{ hashFiles('**/poetry.lock') }}
         restore-keys: |
-          ${{ runner.os }}-venvs
+          ${{ matrix.os }}-venvs-${{ matrix.python-version }}
+          ${{ matrix.os }}-venvs
```

Comment:
So I barely understand what this is doing, although I am reverse-engineering it in my mind and understand the intent.

With that proviso, should the order of L41 and L42 be reversed? I am assuming that in the case of key conflicts, the later definition would overwrite an earlier one, and presumably we would want any python-version-specific definitions to replace more generic ones...

Basically clueless here.

Reply:
Nope! 😄 
https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action